api/envoy/config/core/v3/protocol.proto

```diff
@@ -177,7 +177,28 @@ message Http1ProtocolOptions {
   google.protobuf.BoolValue override_stream_error_on_invalid_http_message = 7;
 }
 
-// [#next-free-field: 15]
+message KeepaliveSettings {
+  // Send HTTP/2 PING frames at this period, in order to test that the connection is still alive.
+  google.protobuf.Duration interval = 1 [(validate.rules).duration = {
+    required: true
+    gte {nanos: 1000000}
+  }];
+
+  // How long to wait for a response to a keepalive PING. If a response is not received within this
+  // time period, the connection will be aborted.
+  google.protobuf.Duration timeout = 2 [(validate.rules).duration = {
+    required: true
+    gte {nanos: 1000000}
+  }];
+
+  // An optional jitter amount as a percentage of interval. If specified,
+  // during every interval Envoy will add interval *
+  // interval_jitter_percent / 100 to the wait time.
+  // A value of zero/unspecified means there will be no jitter.
+  uint32 interval_jitter_percent = 3;
```

Comment:
I think this should be `type.v3.Percent`, similar to what we do in `api/envoy/extensions/filters/http/adaptive_concurrency/v3/adaptive_concurrency.proto`

Reply:
I didn't know that type existed. I'll switch to it.